### ISSUE TITLE
Request: HULU MSO support

### ISSUE THREAD

**Issue by ahmedtds** [rg3/youtube-dl#14810]
Can you add hulu mso login support please?

**Comment by stinkteeth** [comment]
Will adding MSO support help?  Hulu (except for mobile enabled clips and limited episodic content like Larry King and Charlie Rose) is drm constrained.  And the limited content that is drm-free doesn't require MSO authentication.

Rumor has it that a tool named "WVDownloader" uses a drm-free SmartTV app hole for all hulu content (except for the new live TV content) and for that to work MSO authentication is required.  But in order to add that functionality to youtube-dl, someone who has WVDownloader would need to reverse engineer the network traffic and/or methodology used by WVDownloader.

**Comment by ahmedtds** [comment]
Yeah I know about drm, this request is not for hulu content. It's for only Hulu login support for other sites that you can login with hulu as a tv provider option aka mso login support. They are not drm constrained.

**Comment by Aniruddh-J** [comment]
I would vote for this. If anyone needs account info, I am ready to provide it.

**Comment by poe81** [comment]
> Will adding MSO support help? Hulu (except for mobile enabled clips and limited episodic content like Larry King and Charlie Rose) is drm constrained. And the limited content that is drm-free doesn't require MSO authentication.
> 
> Rumor has it that a tool named "WVDownloader" uses a drm-free SmartTV app hole for all hulu content (except for the new live TV content) and for that to work MSO authentication is required. But in order to add that functionality to youtube-dl, someone who has WVDownloader would need to reverse engineer the network traffic and/or methodology used by WVDownloader.

Sorry to post this here, but i have a question

I keep getting this error:
 ERROR - client - Could not find audio metadata
 ERROR - client - cannot auto select audio language

I dont know what i am doing wrong...could someone please please help me

**Comment by mshiels** [comment]
This would really open up alot of options for an hulu subscribers.  Still need an account??

**Comment by PRiiXX** [comment]
sadly this hasn't been implemented after all those years.. I'd too love being able to login to Hulu as a TV provider to download from A&E..

**Comment by Dalarialus** [comment]
I'd love to be able to use Hulu as an MSO login for other services like NBC.

**Comment by 89z** [comment]
you don't need login for NBC...

**Comment by Dalarialus** [comment]
> you don't need login for NBC...

You do for locked content

**Comment by 89z** [comment]
Example URL?

**Comment by 89z** [comment]
> You do for locked content

No you don't. If you have the NBC-Security key (which is publicly available) then you can access NBC content anonymously.
